Combined PRs and additional fixes

mitchcapper opened the pull request:
This combines the following:

- closes #56 
- closes #58 
- closes #59 
- closes #66 
- closes #84 
- closes #88 
- closes #96 

including many great commits by @flaviozavan, @skashyap7, @ni0ki, @mu-ramadan , @nilpunning, @chadcromwell


all commits were generally left intact so original owners are sighted.  @mu-ramadan thanks for something so comprehensive, I did rollback the major changes to whitespace/formatting/line wrapping you added.  It makes it very hard otherwise to track history verse the current (and merging some of the older PR a bigger challenge).   In terms of line-wrapping editors are far better at line wrapping in most situations and it avoids everyone being forced to a specific column width.  I am aware we already wrap the hard coded file lists, but we don't need to wrap so many additional lines.

I still need to fix some of the updates to support all their new measure features and labeling but this seems to be quite comprehensive on the models I have tested so far.  There also seems to be an issue hosting on non-standard ports for some reason.


I will remove the JS please post a bug message as well the brute option should catch all.

chadcromwell commented:
👍

rebane2001 commented:
Haven't tested, but LGTM otherwise. Feel free to merge when ready.

rebane2001 commented on README.md:
```diff
@@ -7,14 +7,16 @@ A tool to download/archive [Matterport](https://matterport.com) virtual tours.
 -   Add `--advanced-download` to a download run to try and download the needed textures and files for supporting dollhouse/floorplan views.  NOTE: Must use built in webserver to host content for this to work.
-
-
+-   Add `--no-tilda` strip paths that contain '~' to work on other platforms. Note this only works with the internal server.
```
I think this should be tilde not tilda

mitchcapper commented:
Yeah I plan to fix the few remaining things and do several more test runs this weekend.  I need a reinvite to the org I didn't accept in time last time.

mitchcapper commented on .gitignore:
```diff
@@ -0,0 +1,10 @@
+# Ignore everything
```
Agreed, I think I was planning to move dl's to a subfolder I would check the root folder by default for existing but if not new ones would go to a sub folder (and add an arg to change that path)

rebane2001 commented:
Sent invite

mitchcapper commented:
Sorry I have greatly increased the scope of this PR.  I figured if there were some vast changes that everyone should test might as well make them all at once. I got most of the backlog of items on my todo for it done.

The major items include:
- moved from threaded sync to async python, consilidated to one download handler that handles most downloads elegantly (and with progress bars).  Async may be slower than threaded but not sure our speed is a big issue, it also uses far less CPU.  Part of the slower download speed is the additional files we are now able to detect and capture.
- All matterport functionality and plugins work including the new dimensions/floorplan info for all models that support it.
- No longer modify any of the original files, but instead modify filename.modified.html etc that will be automatically served in its place
- Added a javascript proxy engine that rewrites all queries in realtime.  This allows us to not have to change/rewrite/remove the hostnames in the original code (fixing a few bugs) and is more reliable to make sure we get everything (and working for non default listening ports)
- command line handling overhauled, several new options, save the initial CLAs with the run itself.  Also copy the matterport script and proxy script to the run item folder to be able to have them incase there was some breaking change and a newer version did not work
- downloads now go to a sub directory instead (but old hosted root ones still work without being moved)


Have not tested back-compat with old downloads throughly but seems to work fine.

Still need to properly open tileset model files(glb) in binary form and be able to still regex search them without having to re-download each time

The one commit I have not done but will do so right before comitting to trunk is formatting.  I already committed a pyproject and vsworkspace files up to enable ruff formatting.  It should avoid massive format only changes in the future by standardizing everything but has the one time cost of format updates.  You can run the formatter yourself to preview, I mostly emulated the style we already used, but didn't want it further cluttering this PR up.

## Overall please test find models that are broken.



I am excited about the potential future of the proxy script.  Two big things:
1. restore the ability to host models on standard web servers without any fancy rules.  It can change post request to get, and redirect things as we want so could make that a very real possibility
2. Future compatability fixes and query updates could be (mostly) automated.  It can be injected into the live site and log all things to a local service in the future to avoid us having to figure out what is different manually.
